src/libs/Navigation/NavigationRoot.js

```diff
@@ -48,26 +48,32 @@ function parseAndLogRoute(state) {
     } else {
         Log.info('Navigating to route', false, {path: currentPath});
     }
-
-    Navigation.setIsNavigationReady();
 }
 
 function NavigationRoot(props) {
     useFlipper(navigationRef);
     const navigationStateRef = useRef(undefined);
+    const onReadyCalled = useRef(false);
 
     const updateSavedNavigationStateAndLogRoute = (state) => {
         navigationStateRef.current = state;
         props.updateCurrentReportId(state);
         parseAndLogRoute(state);
+
+        // NavigationContainer's onReady callback is unreliable (it will fire but navigationRef.isReady() will still be false)
+        // So we use this onStateChange callback instead
```

Comment:
I'm finding `onReady` for `NavigationContainer` is still unreliable even after the navigation reboot. As I've seen in the past, `onStateChange` is being a lot more reliable. Thoughts? cc @marcaaron, @vitHoracek, @techievivek, @adamgrzybowski

Reply:
Well nvm onStateChange is not working on first start after a fresh install. Looking for a better solution to this

Reply:
Alright I was able to fix this without changing our onReady callbacks though they are a bit "off" still